lib/puppet/provider/rsync/rsync.rb

```diff
@@ -159,6 +159,8 @@ def get_target
 
   def get_flags
     flags = []
+    flags << '-p' if @resource.preserve_perms?
+    flags << '--chmod=ugo=rwX' if not @resource.preserve_perms?
```

Comment:
Generally, everything here looks OK, but we can't do this by default.

A `chmod` parameter should probably be added to set this explicitly but never be set by default. There's simply no "good" setting for a default chmod that's not excessively restrictive.

I could probably see doing `--chmod=u=rwX,g=rX,o-rwx` as a sane default but it seems like making this explicit would be better.

Reply:
The `--chmod` would only be triggered on setting preserve_perms to false, which wouldn't be the default.

It looked like in order to preserve permissions on the top-level destination folder, I had to apply all of those changes beyond avoiding the `-p` flag. That included acls, chmod, and xattrs. Otherwise, when rsyncing from my mirror site to the PXE server, puppet would set the top-level linux-install folder to mode 750, and rsync would set it back to 755. So there would be constant flipping of permissions every time the agent ran.

If I did something wrong in testing and my system's observed behavior can't be replicated, I'll adjust the PR down to something smaller.

Reply:
Yeah, you'll need to have rsync match the puppet permissions to not get flapping. This is used for more than tftp so we need to make sure that it's generically safe.